pwiz_tools/Skyline/Controls/Graphs/AllChromatogramsGraph.cs

```diff
@@ -497,8 +497,9 @@ private void AddProgressControls(MultiProgressStatus status)
 
             // Match each file status with a progress control.
             bool first = true;
-            var width = flowFileStatus.Width - 2 -
-                        (flowFileStatus.VerticalScroll.Visible
+            var width = flowFileStatus.Width - 2 - // Avoid clipping the cancel/retry button when we need a vertical scrollbar
+                        (flowFileStatus.VerticalScroll.Visible || 
+                         status.ProgressList.Count > (panelFileList.Height / (new FileProgressControl()).Height)  // If scrollbar isn't visible already, it's about to be

```

Comment:
On further through, I was curious why if status.ProgressList.Count is greater than zero (which it should always be, right?) you need to create a new FileProgessControl?

Reply:
Merely to know the expected height of the ones we'll actually employ. If there's a cleaner way to do that, happy to make the change.

Reply:
Maybe you are right, if they have not been created yet at the time this check is made.

Reply:
I thought about trying to do something clever with the first one that actually gets created and used, but it just didn't seem worth the additional code complexity.